### PR TITLE
refactor: Remove reference to EchoDatabase from lower level database

### PR DESCRIPTION
**Pull request by mykola-vrmchk**
### Motivation / Background

This Pull Request ...

### CODE REVIEW

File: packages/core/echo/echo-db/src/query/filter.ts [hunk 1]
```diff
@@ -188,16 +188,30 @@ export class Filter<T extends {} = any> {
...
+/**
+ * Query logic that checks if object complaint with a filter.
+ * @param echoObject used for predicate filters only.
+ * @returns
+ */
+export const filterMatch = (
+  filter: Filter,
+  core: AutomergeObjectCore | undefined,
+  // TODO(mykola): Remove predicate filters from this level query. Move it to higher proxy level.
+  echoObject?: EchoReactiveObject<any> | undefined,
```
Comment (dmaretskyi) [inline packages/core/echo/echo-db/src/query/filter.ts:200]:
```suggestion
  echoObject: EchoReactiveObject<any> | undefined,
```

File: packages/core/echo/echo-db/src/database.ts [hunk 2]
```diff
@@ -253,6 +264,10 @@ export class EchoDatabaseImpl extends Resource implements EchoDatabase {
    * @deprecated
    */
   get objects(): EchoReactiveObject<any>[] {
+    // Initialize all proxies.
+    for (const core of this._automerge.allObjectCores()) {
```
Comment (dmaretskyi) [inline packages/core/echo/echo-db/src/database.ts:268]: this is way to expensive to be done on this getter

File: packages/core/echo/echo-db/src/database.ts [hunk 3]
```diff
@@ -179,7 +186,7 @@ export class EchoDatabaseImpl extends Resource implements EchoDatabase {
       return undefined;
     }
 
-    const object = defaultMap(this._rootProxies, core, () => initEchoReactiveObjectRootProxy(core));
+    const object = defaultMap(this._rootProxies, core, () => initEchoReactiveObjectRootProxy(core, this));
```
Comment (dmaretskyi) [inline packages/core/echo/echo-db/src/database.ts:189]: extract `defaultMap(this._rootProxies, core, () => initEchoReactiveObjectRootProxy(core, this));`

File: packages/core/echo/echo-db/src/database.ts (hunk 2 above)
Comment (mykola-vrmchk) [reply packages/core/echo/echo-db/src/database.ts:268]: This getter should not be used at all

**Pull request merged**